frontend/src/components/Permissions.tsx

```diff
@@ -1,274 +1,235 @@
+  capabilities: WalletCapabilities;
   initialPermissions: AppPermissions;
   onPermissionsChange: (permissions: AppPermissions) => void;
+  canEditPermissions?: boolean;
   budgetUsage?: number;
-  canEditPermissions: boolean;
   isNewConnection?: boolean;
 }
 
 const Permissions: React.FC<PermissionsProps> = ({
+  capabilities,
   initialPermissions,
   onPermissionsChange,
   canEditPermissions,
   isNewConnection,
   budgetUsage,
 }) => {
   const [permissions, setPermissions] = React.useState(initialPermissions);
-  const [days, setDays] = useState(isNewConnection ? 0 : -1);
-  const [expireOptions, setExpireOptions] = useState(!isNewConnection);
-  const { data: capabilities } = useCapabilities();
 
-  useEffect(() => {
-    setPermissions(initialPermissions);
-  }, [initialPermissions]);
+  const [isScopesEditable, setScopesEditable] = React.useState(
+    isNewConnection ? !initialPermissions.scopes.size : canEditPermissions
+  );
+  const [isBudgetAmountEditable, setBudgetAmountEditable] = React.useState(
+    isNewConnection
+      ? Number.isNaN(initialPermissions.maxAmount)
+      : canEditPermissions
+  );
+  const [isExpiryEditable, setExpiryEditable] = React.useState(
+    isNewConnection ? !initialPermissions.expiresAt : canEditPermissions
+  );
 
-  const handlePermissionsChange = (
-    changedPermissions: Partial<AppPermissions>
-  ) => {
-    const updatedPermissions = { ...permissions, ...changedPermissions };
-    setPermissions(updatedPermissions);
-    onPermissionsChange(updatedPermissions);
-  };
+  // triggered when changes are saved in show app
+  React.useEffect(() => {
+    if (isNewConnection || canEditPermissions) {
+      return;
+    }
+    setPermissions(initialPermissions);
+  }, [canEditPermissions, isNewConnection, initialPermissions]);
 
-  const handleScopeChange = (scope: Scope) => {
-    if (!canEditPermissions) {
+  // triggered when edit mode is toggled in show app
+  React.useEffect(() => {
+    if (isNewConnection) {
       return;
     }
+    setScopesEditable(canEditPermissions);
+    setBudgetAmountEditable(canEditPermissions);
+    setExpiryEditable(canEditPermissions);
+  }, [canEditPermissions, isNewConnection]);
 
-    let budgetRenewal = permissions.budgetRenewal;
+  const [showBudgetOptions, setShowBudgetOptions] = React.useState(
+    isNewConnection ? !!permissions.maxAmount : true
+  );
+  const [showExpiryOptions, setShowExpiryOptions] = React.useState(
+    isNewConnection ? !!permissions.expiresAt : true
+  );
 
-    const newScopes = new Set(permissions.scopes);
-    if (newScopes.has(scope)) {
-      newScopes.delete(scope);
-    } else {
-      newScopes.add(scope);
-      if (scope === "pay_invoice") {
-        budgetRenewal = "monthly";
-      }
-    }
+  const handlePermissionsChange = React.useCallback(
+    (changedPermissions: Partial<AppPermissions>) => {
+      const updatedPermissions = { ...permissions, ...changedPermissions };
+      setPermissions(updatedPermissions);
```

Comment:
Setting local permissions separately as depending on initialPermissions might cause issues due to param data being changed. (Param data is only set from permissions and not initialPermissions so it doesn't change when it is updated, and the isEditable states are not disturbed post that in NewApp screen)